src/bellpepper/r1cs.rs

```diff
@@ -85,29 +86,66 @@ macro_rules! impl_nova_shape {
             &constraint.2,
           );
         }
-
         assert_eq!(num_cons_added, num_constraints);
 
+        A.cols = num_vars + num_inputs;
+        B.cols = num_vars + num_inputs;
+        C.cols = num_vars + num_inputs;
+
         let S: R1CSShape<G> = {
-          // Don't count One as an input for shape's purposes.
```

Comment:
This comment is load bearing for anyone that's not an R1CS expert so I'd like to keep it.

Reply:
💯 As I recall, Nova's R1CS shapes are not always intuitive, and any hints to let readers reconstruct the logic are worth retaining.